ui/src/components/Drawer/index.vue

```diff
@@ -37,99 +30,62 @@ const DRAWER_HEADER_STYLE = {
-  drawerTitle: t('Settings'),
-  items: [
-    {
-      type: 'select',
-      label: `${t('Theme')}:`,
-      labelIcon: Palette,
-      labelStyle: {
-        fontSize: '14px',
-      },
-      showMore: false,
-      value: 'default',
-    },
-    {
-      type: 'list',
-      label: `${t('OnlineUsers')}:`,
-      labelIcon: UsersRound,
-      labelStyle: {
-        fontSize: '14px',
-      },
-    },
-    {
-      type: 'create',
-      label: `${t('CreateLink')}:`,
-      labelIcon: Share2,
-      labelStyle: {
-        fontSize: '14px',
-      },
-      showMore: false,
-    },
-    {
-      type: 'keyboard',
-      label: `${t('Hotkeys')}:`,
-      labelIcon: Keyboard,
-      labelStyle: {
-        fontSize: '14px',
-      },
-    },
-  ],
-});
+// const { t } = useI18n();
 
-const drawerDefaultWidth = computed(() => {
-  return props.contentType === 'setting' ? 502 : 702;
+onMounted(() => {
+  lunaCommunicator.onLuna(LUNA_MESSAGE_TYPE.CHANGE_MAIN_THEME, handleMainThemeChange);
 });
-const disabledFileManager = computed(() => {
-  return FILE_SUFFIX_DATABASE.includes(props.defaultProtocol);
+
+onUnmounted(() => {
+  lunaCommunicator.offLuna(LUNA_MESSAGE_TYPE.CHANGE_MAIN_THEME, handleMainThemeChange);
 });
 
 /**
  * @description 关闭抽屉
  */
-function closeDrawer() {
-  emit('update:open', false);
-}
 
 function handleChangeTab(tab: ContentType) {
   emit('update:content-type', tab);
 }
 </script>
 
 <template>
-  <n-drawer
-    id="drawer-inner-target"
-    resizable
-    placement="right"
-    :show="showDrawer"
-    :min-width="drawerMinWidth"
-    :max-width="drawerMaxWidth"
-    :width="drawerDefaultWidth"
-    @mask-click="closeDrawer"
-  >
+  <n-drawer id="drawer-inner-target" resizable placement="right" :show="true" :show-mask="false" :default-width="600">
     <n-drawer-content closable :native-scrollbar="false" :header-style="DRAWER_HEADER_STYLE">
-      <n-tabs
-        size="small"
-        type="bar"
-        :value="contentType"
-        :pane-style="{ marginTop: '10px' }"
-        @update:value="handleChangeTab"
-      >
-        <n-tab-pane name="setting" display-directive="if" :tab="t('Settings')">
-          <Setting :settings="settingsConfig" />
-        </n-tab-pane>
-        <n-tab-pane
-          name="file-manager"
-          display-directive="show"
-          :disabled="disabledFileManager"
-          :tab="t('FileManagement')"
-        >
-          <FileManager :sftp-token="token" />
+      <n-tabs size="small" type="segment" :pane-style="{ marginTop: '10px' }" @update:value="handleChangeTab">
+        <n-tab-pane v-for="tab in drawerTabs" :key="tab.name" :name="tab.name">
+          <template #tab>
+            <n-flex align="center">
+              <component :is="tab.icon" :size="16" />
+              <span>{{ tab.label }}</span>
+            </n-flex>
+          </template>
+
+          <component :is="tab.component" />
         </n-tab-pane>
       </n-tabs>
     </n-drawer-content>
```

Comment:
There aren't any known issues, irregularities, or performance bottlenecks to report at this time. The template code appears well formed and follows best practices.

The only minor detail to consider is that we should ensure our TypeScript types adhere to the latest versions as they can cause problems if not updated correctly in newer Vue versions since some of them might be incompatible with outdated ones used elsewhere. In terms of optimizing, it's hard without having access to live production data. However, some tips that could potentially boost performance would include reducing unnecessary component instances, minifying your CSS, JavaScript files, ensuring components are properly optimized and reworked when necessary, etc.